File: in_Java/0013 Roman to Integer.java
class Solution {
    public int romanToInt(String s) {
        Map<String, Integer> hm = new HashMap<>();
        String[] arr1 = new String[]{"I","V","X","L","C","D","M","IV","IX","XL","XC","CD","CM"};
        int[] arr2 = new int[]{1,5,10,50,100,500,1000,4,9,40,90,400,900};
        for(int i = 0; i < arr1.length; i++){
            hm.put(arr1[i],arr2[i]);
        }
        int res = 0;
        for(int i = 0; i < s.length(); ){
            if(i+1 < s.length() && hm.containsKey(s.substring(i,i+2))){
                res += hm.get(s.substring(i,i+2));
                i += 2;
            }else{
                res += hm.get(s.substring(i,i+1));
                i++;
            }
        }
        return res;
    }
}

File: in_Python/0013 Roman to Integer.py
class Solution(object):
    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """
        d = {}
        d["I"] = 1
        d["V"] = 5
        d["X"] = 10
        d["L"] = 50
        d["C"] = 100
        d["D"] = 500
        d["M"] = 1000
        d["IV"] = 4
        d["IX"] = 9
        d["XL"] = 40
        d["XC"] = 90
        d["CD"] = 400
        d["CM"] = 900
        res = 0
        i = 0
        while i < len(s)-1:
            s1 = s[i]
            s2 = s[i:i+2]
            if s2 in d:
                res += d[s2]
                i += 2
            else:
                res += d[s1]
                i += 1
        if i < len(s):
            res += d[s[i]]
        return res